sql/log.cc

```diff
@@ -2082,6 +2082,18 @@ bool MYSQL_QUERY_LOG::write(THD *thd, ulonglong current_utime,
 #if defined(ENABLED_PROFILING)
     thd->profiling.print_current(&log_file);
 #endif
+    if ((thd->variables.log_slow_verbosity & (1ULL << SLOG_V_MICROTIME))
+        && thd->trx_start_utime)
+    {
+        char buf[30];
+        snprintf(buf, 30, "%lX%llX", thd->thread_id, thd->trx_start_utime);
+        if (!(thd->server_status & SERVER_STATUS_IN_TRANS))
+            thd->trx_start_utime= 0;
+        if (my_b_printf(&log_file,
+                    "# Trx_id: %s\n", buf) == (uint) -1)
```

Comment:
We end up with two trx ids, innodb one, and thread id. Probably it's better to implement the InnoDB trx id for "COMMIT", and nothing for "BEGIN" (maybe in some cases BEGIN gets InnoDB trx id - what about "WITH CONSISTENT SNAPSHOT"?)